app/application/services/document_category_metadata_validator.py

```
@@ -0,0 +1,50 @@
+import jsonschema
+
+from app.domain.exceptions import SchemaValidationException
+
+if TYPE_CHECKING:
+    from app.application.interfaces.repositories import IDocumentCategoryRepository
+
+
+class DocumentCategoryMetadataValidator:
+    """Validates document metadata against document category metadata_schema when category is configured."""
+
+    def __init__(self, category_repo: "IDocumentCategoryRepository") -> None:
```

Ruff (0.15.0)

[warning] Remove quotes from type annotation

Remove quotes

(UP037)